statistics/histogram.go

```diff
@@ -588,6 +576,23 @@ func (hg *Histogram) Copy() *Histogram {
 	return &newHist
 }
 
+// RemoveUpperBound removes the upper bound from histogram.
+// It is used when merge stats for incremental analyze.
+func (hg *Histogram) RemoveUpperBound() *Histogram {
+	hg.Buckets[hg.Len()-1].Count -= hg.Buckets[hg.Len()-1].Repeat
+	hg.Buckets[hg.Len()-1].Repeat = 0
+	return hg
+}
+
+// TruncateHistogram truncates the histogram to `numBkt` buckets.
+func (hg *Histogram) TruncateHistogram(numBkt int) *Histogram {
+	hist := hg.Copy()
+	hist.Buckets = hist.Buckets[:numBkt]
+	hist.Bounds.TruncateTo(numBkt * 2)
+	hist.NDV = int64(float64(hg.NDV) * (hist.TotalRowCount() / hg.TotalRowCount()))
```

Comment:
For PK, `hist.NDV` should be `hist.TotalRowCount()` now; for normal index, feedback would not update NDV of Histogram according to `UpdateHistogram`, so `hist.NDV` should not be modified either here?